python-api/livy/client.py

```diff
@@ -0,0 +1,444 @@
+        the submitted job in the same thread processing the RPC message,
+        so that queueing does not apply.
+
+        It's recommended that this method only be used to run code that
+        finishes quickly. This avoids interfering with the normal operation
+        of the context.
+
+        Parameters
+        ----------
+        job : function
+            The function must accept a single parameter, which is an instance
+            of JobContext. Spark jobs can be created with the help of
+            JobContext, which exposes the Spark libraries.
+
+        Returns
+        -------
+        future : concurrent.futures.Future
+            A future to monitor the status of the job
+
+        Examples
+        -------
+        >>> def simple_spark_job(context):
+        >>>     elements = [10, 20, 30, 40, 50]
+        >>>     return context.sc.parallelize(elements, 2).count()
+
+        >>> client.run(simple_spark_job)
+        """
+        return self._send_job("run-job", job)
+
+    def add_file(self, file_uri):
+        """
+        Adds a file to the running remote context.
+
+        Note that the URL should be reachable by the Spark driver process. If
+        running the driver in cluster mode, it may reside on a different
+        host, meaning "file:" URLs have to exist on that node (and not on
+        the client machine).
+
+        Parameters
+        ----------
+        file_uri : pathlib.PurePath or pathlib2.PurePath, string in the format
+            of file uri
+        Representation of the path to a local file using filesystem encoding.
+
+        Returns
+        -------
+        future : concurrent.futures.Future
+            A future to monitor the status of the job
+
+        Examples
+        -------
+        >>> import pathlib2
+        >>> file_uri = pathlib2.Path("/test_add.txt").as_uri()
+
+        >>> client.add_file(file_uri)
+
+        >>> # Example job using the file added using add_file function
+        >>> def add_file_job(context):
+        >>>    from pyspark import SparkFiles
+        >>>    def func(iterator):
+        >>>        with open(SparkFiles.get("test_add.txt")) as testFile:
+        >>>        fileVal = int(testFile.readline())
+        >>>        return [x * fileVal for x in iterator]
+        >>>    return context.sc.parallelize([1, 2, 3, 4])
+        >>>             .mapPartitions(func).collect()
+
+        >>> client.submit(add_file_job)
+        """
+        return self._add_file_or_pyfile_job("add-file", file_uri)
+
+    def add_jar(self, file_uri):
+        """
+        Adds a jar file to the running remote context.
+
+        Note that the URL should be reachable by the Spark driver process. If
+        running the driver  in cluster mode, it may reside on a different host,
+         meaning "file:" URLs have to exist on that node (and not on the
+         client machine).
+
+        Parameters
+        ----------
+        file_uri : string, pathlib.PurePath or pathlib2.PurePath
```

Comment:
I'm not a fan of requiring an external library (`pathlib*`) for something that doesn't really require it. A string here should be enough, right? The server will complain if it's wrong.